lib/LDKeyPair.js

```diff
@@ -45,21 +51,58 @@ class LDKeyPair {
     throw new Error('Abstract method, must be implemented in subclass.');
   }
 
+  /**
+   * Imports a key pair instance from a provided externally fetched key
+   * document (fetched via a secure JSON-LD `documentLoader` or via
+   * `cryptoLd.fromKeyId()`), optionally checking it for revocation and required
+   * context.
+   *
+   * @param {object} options - Options hashmap.
+   * @param {string} options.document - Externally fetched key document.
+   * @param {boolean} [options.checkContext=true] - Whether to check that the
+   *   fetched key document contains the context required by the key's crypto
+   *   suite.
+   * @param {boolean} [options.checkRevoked=true] - Whether to check the key
+   *   object for the presence of the `revoked` timestamp.
+   *
+   * @returns {Promise<LDKeyPair>} Resolves with the resulting key pair
+   *   instance.
+   */
+  static async fromKeyDocument({document, checkContext = true,
+    checkRevoked = true
+  } = {}) {
+    if(!document) {
+      throw new TypeError('The "document" parameter is required.');
+    }
+
+    if(checkContext) {
+      const fetchedDocContexts = [].concat(document['@context']);
+      if(!fetchedDocContexts.includes(this.SUITE_CONTEXT)) {
```

Comment:
Hmm, this is a static method so `this` will be the `globalThis` (or whatever someone sets it to). So I don't think this works properly, does it? This gets back to the weird ergonomics issue. I think it can only really start to be solved by having `.use` gather all of the suite contexts and have it check for N matching suites based on the context found in the key document -- and then differentiate based on `type` from there ... based on what the `type` term is given that context. I think that's too messy for now ... so we should just put this back.

Unless somehow it works in a way I don't understand ... or if `static` can just be removed.

Reply:
Oh, I see that `this` gets set to the class itself. So this does work. Ok, that's not too terrible then. Thanks JS!

Reply:
heheheh yep, was just about to say.